src/dnx.clr.managed/DomainManager.cs

```diff
@@ -36,12 +37,34 @@ public override void InitializeNewDomain(AppDomainSetup appDomainInfo)
             Environment.SetEnvironmentVariable(EnvironmentNames.AppBase, _info.ApplicationBase);
         }
 
-        var version = DetermineDnxVersion(_info.ApplicationBase);
+        appDomainInfo.ApplicationBase = Environment.GetEnvironmentVariable(EnvironmentNames.DefaultLib);
+        appDomainInfo.TargetFrameworkName = DetermineAppDomainTargetFramework();
+    }
+
+    private string DetermineAppDomainTargetFramework()
+    {
+        // Check if the native bootstrapper gave us a framework name
+        string frameworkName = Environment.GetEnvironmentVariable(EnvironmentNames.Framework);
```

Comment:
Yes, see #1970 - We are aware the environment variable is bad, but until there's a better system this is the way we've got.